### PR TITLE
#86322 Replace TinyMCE Popup with Thickbox Modal

### PR DESCRIPTION
**Pull request by ggwicz**
_Ref:_ [C#86322](http://central.tri.be/issues/86322)

### CODE REVIEW

File: includes/pardot-forms-shortcode-popup-class.php [hunk 1]
```diff
@@ -161,154 +101,38 @@ function get_body_inner_html() {
...
+		$formsec        = __( 'Forms', 'pardot' );
+		$labelform      = __( 'Select a form to insert', 'pardot' );
+		$formcust       = __( 'Optional iframe Parameters', 'pardot' );
+        $dccust         = __( 'Optional iframe Parameters', 'pardot' );
+		$labelformh     = __( 'Height', 'pardot' );
+		$labelformw     = __( 'Width', 'pardot' );
+		$labelformc     = __( 'Class', 'pardot' );
+        $labeldch       = __( 'Height', 'pardot' );
+        $labeldcw       = __( 'Width', 'pardot' );
+        $labeldcc       = __( 'Class', 'pardot' );
```
Comment (bordoni) [inline includes/pardot-forms-shortcode-popup-class.php]: Adjust the Indentation.

File: includes/pardot-forms-shortcode-popup-class.php [hunk 2]
```diff
@@ -161,154 +101,38 @@ function get_body_inner_html() {
...
-}
-</script>
-HTML;
+        $dcparam_text   = __( 'Height and width should be in px or % (i.e. 250px or 90%).', 'pardot' );
+
+        ob_start();
+
+        include PARDOT_PLUGIN_DIR . '/includes/popup.php';
+
+        $html = ob_get_clean();
```
Comment (bordoni) [inline includes/pardot-forms-shortcode-popup-class.php]: Adjust indentation

File: includes/pardot-plugin-class.php [hunk 3]
```diff
@@ -892,9 +897,9 @@ static function get_dynamic_content_body( $args = array() ) {
 				/**
 				 * Use the dynamicContent_id to find the right one
 				 */
-				$dynamicContent = $dynamicContents[$dynamicContent_id];
-				$dynamicContent_html = $dynamicContent->embedCode;
-                $dynamicContent_url = $dynamicContent->embedUrl;
+				$dynamicContent         = $dynamicContents[ $dynamicContent_id ];
+				$dynamicContent_html    = $dynamicContent->embedCode;
+                $dynamicContent_url     = $dynamicContent->embedUrl;
```
Comment (bordoni) [inline includes/pardot-plugin-class.php]: Adjust the indentation

File: includes/popup.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<div id="pardot-forms-shortcode-insert-dialog">
+
+	<form method="post"	action="#">
```
Comment (bordoni) [inline includes/popup.php]: remove indent from here.

File: README.txt [hunk 5]
```diff
@@ -140,6 +140,10 @@ Filter the regular expression used to find URLs to be converted to https://go.pa
 
 == Changelog ==
 
+= 1.4.4 =
+
+* Fix - Restore functionality of the Pardot shortcode builder. It now relies more on Thickbox than on TinyMCE's popup library, which is very outdated. [86322]
```
Comment (barryhughes) [inline README.txt]: This isn't inaccurate, but here in our public changelog I feel we should emphasize the positives and drop the note about the former implementation being outdated. What do you think about something slightly simpler, for example:

> Fix - Restores functionality of the Pardot button in the visual editor [86322]

File: css/popup.css [hunk 6]
```diff
@@ -0,0 +1,59 @@
+/**
+ *
```
Comment (barryhughes) [inline css/popup.css]: Throughout this file: do we need the extra line breaks padding the top and bottom of the multiline comments?

File: includes/popup.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+				<tbody>
+
+					<!-- Selection for "Forms" -->
+					<tr><td><label for="shortcode"><?php echo $labelform; ?></label>:</td></tr>
+
+					<tr>
+						<td>
+							<span id="pardot-forms-shortcode-select">
+								<input type="hidden" id="shortcode">
+								<img class="spinner" src="<?php echo $spinner_url; ?>" height="16" weight="16" alt="<?php esc_attr_e( 'Time waits for no man.', 'pardot' ); ?>">
```
Comment (barryhughes) [inline includes/popup.php]: Should probably escape the `$spinner_url`

File: includes/popup.php [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+							</span>
+						</td>
+					</tr>
+
+					<!-- Optional iframe Parameters -->
+					<tr><td><h4><?php echo $formcust; ?></h4></td></tr>
```
Comment (barryhughes) [inline includes/popup.php]: Ditto throughout: we probably ought to escape this. If we do it here, then it's really clear that it's being escaped - but alternatively we could escape at source by switching from `__()` to `esc_html__()`

File: js/popup.js [hunk 9]
```diff
@@ -0,0 +1,194 @@
+(function( $ ) {
+	// We need to wait for window.load because we're manipulating Thickbox elements.
+	$(window).on( 'load', function() {
```
Comment (barryhughes) [inline js/popup.js]: Should we be adding spaces in the parentheses (ie, `$( window )`) here?

File: README.txt (hunk 5 above)
Comment (ggwicz) [reply README.txt]: great point -- i wasn't thinking of the negative, i was thinking of providing an explanation for folks who might notice a slightly-different appearance of the builder.

i think your suggestion is much friendlier

**Comment by ggwicz** [comment]
great feedback guys, thank you. i've implemented it all and this one's ready for another review

**Pull request merged**